sumologic/resource_sumologic_cse_match_list.go

```diff
@@ -0,0 +1,284 @@
+
+	var i []map[string]interface{}
+
+	for _, t := range items {
+		mapping := map[string]interface{}{
+			"id":          t.ID,
+			"description": t.Meta.Description,
+			"expiration":  t.Expiration,
+			"value":       t.Value,
+		}
+		i = append(i, mapping)
+	}
+
+	d.Set("items", i)
+
+}
+
+func resourceSumologicCSEMatchListDelete(d *schema.ResourceData, meta interface{}) error {
+	c := meta.(*Client)
+
+	//Match list items
+	itemsData := d.Get("items").([]interface{})
+	var items []CSEMatchListItemPost
+	for _, data := range itemsData {
+		items = append(items, resourceToCSEMatchListItem([]interface{}{data}))
+	}
+
+	if len(items) > 0 {
+		for _, item := range items {
+			err2 := c.DeleteCSEMatchListItem(item.ID)
+			if err2 != nil {
+				log.Printf("[WARN] An error occurred while updating match list item wiht id: %s, err: %v", item.ID, err2)
+			}
+		}
+	}
+
+	err := c.DeleteCSEMatchList(d.Id())
+	return err
+
+}
+
+func resourceSumologicCSEMatchListCreate(d *schema.ResourceData, meta interface{}) error {
+	c := meta.(*Client)
+
+	if d.Id() == "" {
+		id, err := c.CreateCSEMatchList(CSEMatchListPost{
+			Active:       true,
+			DefaultTtl:   d.Get("default_ttl").(int),
+			Description:  d.Get("description").(string),
+			Name:         d.Get("name").(string),
+			TargetColumn: d.Get("target_column").(string),
+		})
+
+		if err != nil {
+			return err
+		}
+		d.SetId(id)
+
+		//Match list items
+		itemsData := d.Get("items").([]interface{})
+		var items []CSEMatchListItemPost
+		for _, data := range itemsData {
+			items = append(items, resourceToCSEMatchListItem([]interface{}{data}))
+		}
+
+		if len(items) > 0 {
+			err2 := c.CreateCSEMatchListItems(items, id)
+			if err2 != nil {
+				log.Printf("[WARN] An error occurred while adding match list items to match list id: %s, err: %v", id, err2)
+			}
+
+			// Calling Sleep method, adding items might take a while before items retrieved in next section
+			// Need to find a better way since feels super hacky
+			time.Sleep(15 * time.Second)
+
+		}
+
+	}
+
+	return resourceSumologicCSEMatchListRead(d, meta)
+}
+
+func resourceToCSEMatchListItem(data interface{}) CSEMatchListItemPost {
+	itemsSlice := data.([]interface{})
+	item := CSEMatchListItemPost{}
+	if len(itemsSlice) > 0 {
+		itemObj := itemsSlice[0].(map[string]interface{})
+		item.ID = itemObj["id"].(string)
+		item.Description = itemObj["description"].(string)
+		item.Active = true
+		item.Expiration = itemObj["expiration"].(string)
+		item.Value = itemObj["value"].(string)
+	}
+	return item
+}
+
+func resourceSumologicCSEMatchListUpdate(d *schema.ResourceData, meta interface{}) error {
+	CSEMatchListPost, err := resourceToCSEMatchList(d)
+	if err != nil {
+		return err
+	}
+
+	c := meta.(*Client)
+	if err = c.UpdateCSEMatchList(CSEMatchListPost); err != nil {
+		return err
+	}
+
+	//Match list items
+	itemsData := d.Get("items").([]interface{})
+	var items []CSEMatchListItemPost
+	for _, data := range itemsData {
```

Comment:
This will add new items, but we also need to remove any existing items that were removed from TF

Reply:
thanks, done under https://github.com/SumoLogic/terraform-provider-sumologic/commit/dda976e0c3b3134f2ed3736bdaa8a69b129a6609